extensions/replication/tasks/MultipleBackendTask.js

```diff
@@ -372,14 +334,30 @@ class MultipleBackendTask extends ReplicateObject {
 
     /**
      * Get byte ranges for an object of the given content length, such that the
-     * range count does not exceed 1024 elements.
+     * range count does not exceed 1024 parts if replicating to GCP or does not
+     * exceed 10000 parts otherwise. This method also optimizes for the
+     * subsquent range requests by returning ranges that are sized as powers of
+     * two (aside from the last part).
      * @param {Number} contentLen - The content length of the whole object
+     * @param {Boolean} isGCP - Whether the object is being replicated to GCP
      * @return {Array} The array of byte ranges.
      */
-    _getRanges(contentLen) {
-        // Creates ranges that have sizes that are powers of 2.
+    _getRanges(contentLen, isGCP) {
+        // 5MB min part size rounded up to the nearest power of two is 8388608.
+        let partCount = Math.pow(2,
```

Comment:
This works for now but we need a more efficient way so that we can maximize the usage of 10K parts.

Reply:
Not sure what is the benefit of ensuring the number of parts is a power of two? Something that may benefit from being a power of two instead (or being divisible by a sufficiently large power-of-two number) could be the number of bytes per part, since a lot of layers in our systems tend to perform better when processing data aligned with disk or memory pages, generally speaking.

Usually using a fixed default range size that can be optionally tuned works well in practice, and we should increase the range size if needed so that we stay under 10K ranges.

Reply:
I like this idea @jonathan-gramain, I'll work on implementing that.